schnell/mapping.py

```diff
@@ -687,3 +801,128 @@ def get_G_ell(self, t, f, nside, no_autos=False, deltaOmega_norm=True,
             if np.ndim(t) == 0:
                 gls = np.squeeze(gls, axis=1)
         return gls
+
+
+
+
+
+    def get_N_ell_inv_part(self, t, f, nside, indices, is_fspacing_log=False,
+                  no_autos=False, deltaOmega_norm=True, proj=None):
+        """
+        Choose which parts of the sum to remove
+        """
+        t_use = np.atleast_1d(t)
+        f_use = np.atleast_1d(f)
+        if is_fspacing_log:
+            dlf = np.mean(np.diff(np.log(f)))
+            df = f * dlf
+        else:
+            df = np.mean(np.diff(f)) * np.ones(len(f))
+        gls = self.get_G_ell_part(t_use, f_use, nside, indices, no_autos=no_autos,
+                             deltaOmega_norm=deltaOmega_norm, proj=proj)
+        # Sum over frequencies
+        gls = np.sum(gls * df[:, None, None], axis=0)
+        # Sum over times
+        if len(t_use) == 1:
+            gls = np.squeeze(gls * t)
+        else:
+            dt = np.mean(np.diff(t_use))
+            gls = np.sum(gls, axis=0) * dt
+        return gls
+
+
+    def get_G_ell_part(self, t, f, nside, indices, no_autos=False, deltaOmega_norm=True,
```

Comment:
What are these two functions supposed to do exactly?

Reply:
They just allowed me to compute a part of the sum over the detectors in Nell^-1 and were mainly for checking purposes ; maybe I should delete them